### PR TITLE
Option to emit (but still filter) all germline sites in Mutect2

### PR DESCRIPTION
**Pull request by davidbenjamin**
@takutosato I made sure that when the new option is turned off, as it is by default, nothing changes.  Also, the new option surprisingly only added 30% or so to runtime because M2 already calls so many 
 false active regions.

@chandrans @sooheelee This is the feature we have been discussing.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java [hunk 1]
```diff
@@ -222,6 +220,14 @@ private void addGenotypes(final LikelihoodMatrix<Allele> tumorLog10Matrix,
         callVcb.genotypes(genotypes);
     }
 
+    public static double[] getEffectiveCounts(final LikelihoodMatrix<Allele> log10LikelihoodMatrix) {
+        if (log10LikelihoodMatrix.numberOfReads() == 0) {
+            return new double[log10LikelihoodMatrix.numberOfAlleles()]; // zero couns for each allele
```
Comment (takutosato) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java]: couns -> counts

**Comment by sooheelee** [comment]
Thanks for including us in the discussion @davidbenjamin.

> the new option surprisingly only added 30% or so to runtime because M2 already calls so many
false active regions.

This is a great value-add.

**Comment by sooheelee** [comment]
Just to clarify @davidbenjamin @takutosato, the normal sample calling uses:

1. HaplotypeCaller logic for _germline_ calling OR
2. Mutect2 sensitive calling?

**Comment by davidbenjamin** [comment]
@sooheelee This new option isn't doing germline calling per se.  Rather it calls all variants, somatic and germline, in the tumor sample.  To directly call germline events in the normal it wold be faster and better to use HaplotypeCaller.

**Comment by sooheelee** [comment]
Thanks for the clarification. To paraphrase the following:

> This new option isn't doing germline calling per se. Rather it calls all variants, somatic and germline, in the tumor sample. 

We now have a parameter that allows inclusion of sites that are variant in the tumor that are also variant in the matched-normal and/or present in the germline resource. With the option enabled, these sites are emitted for the tumor and if the matched-normal is part of the analysis, then its variants for the sites are also noted in the callset. Finally, this parameter is set to true with `--genotype-germline-sites`.

**Comment by davidbenjamin** [comment]
That's right, and just because more explicitness never hurts, here's another paraphrase.

M2 default: call somatic variation in the tumor sample.  Some possible germline variants will be included, but not intentionally.

New option: call *all* variation in the tumor sample.

In both case FilterMutectCalls is the same.  It uses the INFO fields from Mutect2 to put the `germline_risk` in the filter field.  If a user really wants to use Mutect2 as a slower HaplotypeCaller she could always set the germline posterior probability threshold to 1, effectively turning off this filter.

**Pull request merged**